verilog/parser/verilog.y

```diff
@@ -3390,12 +3390,26 @@ type_declaration
     { $$ = MakeTaggedNode(N::kTypeDeclaration, $1, $2, $3, $4, $5); }
   | TK_typedef interface_type GenericIdentifier ';'
     { $$ = MakeTaggedNode(N::kTypeDeclaration, $1, $2, nullptr, $3, $4, nullptr); }
+  /* TODO: Figure out how to make the braced members list optional
+     to make these more robust */
   | TK_typedef TK_enum GenericIdentifier ';'
-    { $$ = MakeTaggedNode(N::kTypeDeclaration, $1, $2, nullptr, $3, $4, nullptr); }
+    { $$ = MakeTaggedNode(N::kTypeDeclaration, $1,
+                          MakeTaggedNode(N::kDataTypePrimitive,
+                                         MakeTaggedNode(N::kEnumDataType,
+                                                        $2, nullptr, nullptr)),
+                          nullptr, $3, $4, nullptr); }
```

Comment:
This is good.  Later we can introduce helper functions to constrain the construction of some of these re-used nodes in CST/*.  No further change needed.